go/enclave/vkhandler/vk_handler.go

```diff
@@ -25,9 +25,10 @@ type VKHandler struct {
 // New creates a new viewing key handler
 // checks if the signature is valid
 // as well if signature matches account address
+// todo (@ziga) - function now accepts both old and new messages
 func New(requestedAddr *gethcommon.Address, vkPubKeyBytes, accountSignatureHexBytes []byte) (*VKHandler, error) {
 	// Recalculate the message signed by MetaMask.
-	msgToSign := viewingkey.GenerateSignMessage(vkPubKeyBytes)
+	msgToSign := viewingkey.GenerateSignMessageOG(vkPubKeyBytes, requestedAddr)
```

Comment:
Wont' this return err if it's the old message type ?

Reply:
No, I've tested both cases in the tests.
(We receive address, pubKey and signature)
Message affects just a hash and wrong address is returned in that case a few lines below.